tests/integration/conftest.py

```diff
@@ -96,17 +104,35 @@ async def kube_config_cluster_payload(kube_config_payload: dict[str, Any]) -> An
 def cert_authority_data_pem(
     kube_config_cluster_payload: dict[str, Any]
 ) -> Optional[str]:
-    ca_path = kube_config_cluster_payload["certificate-authority"]
-    if ca_path:
-        return Path(ca_path).read_text()
+    if "certificate-authority" in kube_config_cluster_payload:
+        ca_path = kube_config_cluster_payload["certificate-authority"]
+        if ca_path:
+            return Path(ca_path).read_text()
+    elif "certificate-authority-data" in kube_config_cluster_payload:
+        return base64.b64decode(
+            kube_config_cluster_payload["certificate-authority-data"]
+        ).decode()
     return None
 
 
 @pytest.fixture(scope="session")
 async def kube_config_user_payload(kube_config_payload: dict[str, Any]) -> Any:
+    import tempfile
+
     user_name = "minikube"
     users = {user["name"]: user["user"] for user in kube_config_payload["users"]}
-    return users[user_name]
+    user = users[user_name]
+    if "client-certificate-data" in user:
+        with tempfile.NamedTemporaryFile(delete=False) as cert_file:
+            cert_file.write(base64.b64decode(user["client-certificate-data"]))
+            cert_file.flush()
+            user["client-certificate"] = cert_file.name
+    if "client-key-data" in user:
+        with tempfile.NamedTemporaryFile(delete=False) as key_file:
+            key_file.write(base64.b64decode(user["client-key-data"]))
+            key_file.flush()
+            user["client-key"] = key_file.name
+    return user
```

Comment:
Won't temporary files be destroyed files as soon as you close them?

Reply:
They won't as I pass `delete=False` to both constructors.